app/main.py

```
@@ -1,6 +1,29 @@
+import requests
+from dotenv import load_dotenv
+import os
+
+CITY = "Paris"
+URL = "http://api.weatherapi.com/v1/current.json"
+
+
 def get_weather() -> None:
-    # write your code here
-    pass
+    load_dotenv()
+    api_key = os.getenv("API_KEY")
+    if not api_key:
+        raise Exception("API_KEY environment variable not set")
+
+    params = {"key": api_key, "q": CITY}
+
+    response = requests.get(URL, params=params)
+    data = response.json()
+    if response.status_code != 200:
```

The code checks `response.status_code != 200` after json() but that check comes too late to avoid parsing errors. Move status checking earlier (before parsing) or use `response.raise_for_status()` to fail fast with a clear error.